core/templates/dev/head/pages/exploration-editor-page/statistics-tab/services/learner-action-render.service.ts

```diff
@@ -164,31 +164,39 @@ angular.module('oppia').factory('LearnerActionRenderService', [
     };
 
     /**
-     * Helper object to maintain the status of different display blocks while
-     * splitting up learner actions. This object will be updated as learner
-     * actions are inserted.
+     * Helper object to maintain the status of starting indices while iterating
+     * through the learner actions. This object will be updated as learner
+     * actions are processed.
      */
-    var groupedDisplayBlocks = {
-      displayBlocks: null,
-      localBlock: null,
+    var groupedStartingIndices = {
```

Comment:
Hmm, I think this complicates things since the code base is in the process of moving towards typescript. Could we make this into a function again now that the logic has become simpler?

Reply:
@brianrodri We use a helper object because the member variables involved keep changing. If we pass it straight up to functions, we need to continuously keep track of changed values inside the function and reflect it (since there is no pass by reference in JS). The other alternative is to keep all the functionality without using helper functions at all, which increases code complexity. Cant we define helper objects in Typescript?

Reply:
Let's make it a new domain object, then?